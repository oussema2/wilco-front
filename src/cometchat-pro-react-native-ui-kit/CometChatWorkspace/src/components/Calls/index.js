export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
