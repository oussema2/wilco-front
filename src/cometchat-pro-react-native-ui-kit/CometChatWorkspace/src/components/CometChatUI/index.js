/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
