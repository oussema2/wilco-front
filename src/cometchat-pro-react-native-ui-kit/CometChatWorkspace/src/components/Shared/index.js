export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
