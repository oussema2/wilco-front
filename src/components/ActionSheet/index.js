export { default as ActionSheet } from './ActionSheet';
