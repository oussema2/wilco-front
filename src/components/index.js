export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
