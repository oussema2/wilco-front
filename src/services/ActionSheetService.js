import { makeAutoObservable } from 'mobx';

export default class ActionSheetService {
	constructor() {
		this.isOpen = false;
		this.actions = [];

		makeAutoObservable( this );
	}

	open( config ) {
		this.isOpen = true;
		this.actions = config.actions;
	}

	close() {
		this.isOpen = false;
		this.actions = [];
	}
}
