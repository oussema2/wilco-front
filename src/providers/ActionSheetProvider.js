import React from 'react';
import PropTypes from 'prop-types';
import ActionSheetService from '../services/ActionSheetService';

const ActionSheetContext = React.createContext( null );

export const ActionSheetProvider = ( { children } ) => {
	const actionSheetService = new ActionSheetService();
	return (
		<ActionSheetContext.Provider value={actionSheetService}>
			{children}
		</ActionSheetContext.Provider>
	);
};

ActionSheetProvider.propTypes = {
	children: PropTypes.node.isRequired
};

export const useActionSheetService = () => React.useContext( ActionSheetContext );
