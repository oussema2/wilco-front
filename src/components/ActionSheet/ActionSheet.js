import React from 'react';
import { observer } from 'mobx-react';
import { View } from 'react-native';

import { BottomSheet } from 'react-native-elements';
import { styles } from './ActionSheet.styles';
import { useActionSheetService } from '../../providers/ActionSheetProvider';

import ActionSheetActions from './components/ActionSheetActions';
import ActionSheetCancelButton from './components/ActionSheetCancelButton';

const ActionSheet = () => {
	const actionSheetService = useActionSheetService();

	return (
		<BottomSheet
			isVisible={actionSheetService.isOpen}
			containerStyle={styles.actionSheet}
			modalProps={{
				animationType: 'fade'
			}}
		>
			<View style={styles.container}>
				<ActionSheetActions />
				<ActionSheetCancelButton />
			</View>
		</BottomSheet>
	);
};

export default observer( ActionSheet );
