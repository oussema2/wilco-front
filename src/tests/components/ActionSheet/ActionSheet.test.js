import React from 'react';
import { act, fireEvent, render } from '@testing-library/react-native';
import { ActionSheet } from '../../../components/ActionSheet';
import { ActionSheetProvider } from '../../../providers/ActionSheetProvider';
import mockUseActionSheetService from '../../mocks/mockUseActionSheetService';
import ActionSheetService from '../../../services/ActionSheetService';

describe( '< ActionSheet/>', () => {
	let component;
	const mockOnPress = jest.fn();
	const actionSheetService = new ActionSheetService();
	const config = {
		actions: [ {
			title: 'Test Action', destructive: true, onPress: mockOnPress
		} ]
	};

	const setUp = () => {
		component = render( <ActionSheet />, { wrapper: ActionSheetProvider } );
	};

	const openActionSheet = () => {
		act( () => {
			actionSheetService.open( config );
		} );
	};

	beforeEach( () => {
		jest.clearAllMocks();
		mockUseActionSheetService( { service: actionSheetService } );
		setUp();
	} );

	describe( 'when the actionSheet is closed', () => {
		it( 'renders the ActionSheet as not visible', () => {
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the actionSheet is open', () => {
		it( 'renders the ActionSheet as visible', () => {
			openActionSheet();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when an action button is pressed', () => {
		it( 'calls on press and closes the action sheet', () => {
			openActionSheet();
			fireEvent.press( component.getByText( 'Test Action' ) );
			expect( mockOnPress ).toHaveBeenCalledTimes( 1 );
			expect( actionSheetService.isOpen ).toBe( false );
		} );
	} );

	describe( 'when the cancel button is pressed', () => {
		it( 'closes the action sheet', () => {
			openActionSheet();
			fireEvent.press( component.getByText( 'Cancel' ) );
			expect( actionSheetService.isOpen ).toBe( false );
		} );
	} );
} );
