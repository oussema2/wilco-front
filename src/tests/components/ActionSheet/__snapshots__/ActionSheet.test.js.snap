// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ActionSheet/> when the actionSheet is closed renders the ActionSheet as not visible 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <RNCSafeAreaView
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.2)",
        "flex": 1,
        "flexDirection": "column-reverse",
      }
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <View>
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4291295231,
                  4294967295,
                ]
              }
              endPoint={
                Object {
                  "x": 0.9,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              style={
                Object {
                  "borderRadius": 5,
                  "height": "auto",
                  "width": 335,
                }
              }
              testID="linearGradient"
            />
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 8,
                  "marginBottom": 24,
                  "marginTop": 12,
                  "width": 335,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    undefined,
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="action-sheet-button"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#2089dc",
                        "borderRadius": 0,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "height": 52,
                        "justifyContent": "center",
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#5B5B5B",
                          "fontFamily": "NotoSansJP-Bold",
                          "fontSize": 15,
                          "fontWeight": undefined,
                          "letterSpacing": 0.4,
                          "lineHeight": 20,
                          "paddingVertical": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Cancel
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`< ActionSheet/> when the actionSheet is open renders the ActionSheet as visible 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.2)",
        "flex": 1,
        "flexDirection": "column-reverse",
      }
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <View>
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4291295231,
                  4294967295,
                ]
              }
              endPoint={
                Object {
                  "x": 0.9,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 1,
                  "y": 0,
                }
              }
              style={
                Object {
                  "borderRadius": 5,
                  "height": "auto",
                  "width": 335,
                }
              }
              testID="linearGradient"
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    undefined,
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="action-sheet-button"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#2089dc",
                        "borderRadius": 0,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "height": 52,
                        "justifyContent": "center",
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#5B5B5B",
                          "fontFamily": "NotoSansJP-Bold",
                          "fontSize": 15,
                          "fontWeight": undefined,
                          "letterSpacing": 0.4,
                          "lineHeight": 20,
                          "paddingVertical": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Test Action
                    </Text>
                  </View>
                </View>
              </View>
            </BVLinearGradient>
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 8,
                  "marginBottom": 24,
                  "marginTop": 12,
                  "width": 335,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    undefined,
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="action-sheet-button"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#2089dc",
                        "borderRadius": 0,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "height": 52,
                        "justifyContent": "center",
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#5B5B5B",
                          "fontFamily": "NotoSansJP-Bold",
                          "fontSize": 15,
                          "fontWeight": undefined,
                          "letterSpacing": 0.4,
                          "lineHeight": 20,
                          "paddingVertical": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Cancel
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
